Docs/Specifics/Vision.md

```diff
@@ -2,64 +2,113 @@
+
+![Example of distorted chessboard](/Assets/Distortion.jpg)
+
+It's possible for us to undistort these images—we just need to find the camera-specific numbers that will allow us to perform that math, which we can do by calibrating the camera.
+There are a couple ways to do this, but one way is to take a lot of photos with the camera being calibrated of a chessboard pattern (which we know has straight lines and certain measurements) at different angles and positions and send those photos to programs that then handle the math for us and return those numbers.
+
+### Object Detection
+
+Sometimes we want to be able to detect other things on the field that aren't AprilTags.
+This could include game pieces, other robots, and more.
+One way of accomplishing this is through tuning HSV thresholds to only pick up on a certain color, such as the orange of a Crescendo note.
+Another way is contour filtering, which looks for the outlines of a shape like a circle or a triangle, such as a cone from Charged Up.
+
+<img src="../../Assets/ShapeDetection.webp" alt="Example of detecting a triangular object" width="600"/>
+
+A more complicated system is through machine learning models that have been specially trained on the objects we want to detect, such as game pieces or other robots.
+Due to the significant processing power that this requires, in FRC this is typically done on the Rockchip Neural Processing Unit (NPU), which is on the Orange Pi 5 and similar coprocessors.
+While we haven't implemented this, the greater FRC community has developed several models and other resources.
+In 2024, Photonvision also ships with a pretrained model for notes.
 
 ### PhotonVision
 
-Photonvision is an app and library that gets vision data from the robot for a variety of applications.
-It makes it a lot easier to interface with the camera and also has plenty of utility methods to get information like distance and position, as well as having some built in pose estimator features and more.
+Photonvision is a community, open source library that gets vision data from the robot for a variety of applications.
+It simplifies interfacing with the camera and also has plenty of utility methods to get information like distance and position, as well as having some built in pose estimator features and more.
 
 ### Limelight
 
 Limelight (LL) is a camera and processor system for FRC designed to make vision easy and simple.
-We used a Limelight 2+ with Limelight's software in 2022 for RR tape detection, and a LL2+ with Photonvision software in 2023 for Apriltag detection.
-The LL2+ was not satisfactory for apriltag detection for us, and we will likely move on to different hardware in the near future.
+We used a Limelight 2+ with Limelight's software in 2022 for retroreflective tape detection, and a LL2+ with Photonvision software in 2023 for Apriltag detection.
```

Comment:
add a sentence about LL3 and LL3G such as "Limelight has released the LL3 more recently which posses more processing power for Apriltags and object detection, and other teams have had success with it. The LL3G also features a global shutter camera for further improved Apriltag performance. At time of writing we have not tested either."